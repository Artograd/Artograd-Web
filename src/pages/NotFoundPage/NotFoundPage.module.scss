.wrapper {
  width: 35%;
  margin: 50px auto;
  text-align: center;
  button {
    width: fit-content;
    margin: 30px auto;
  }
}

@media only screen and (max-device-width: 767px) {
  .wrapper {
    width: 90%;
    margin: 100px auto;
  }
}
