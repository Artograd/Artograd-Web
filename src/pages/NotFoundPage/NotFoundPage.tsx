import { Panel, Text, Button } from '@epam/uui';
import styles from './NotFoundPage.module.scss';
import ErrorIcon from './assets/error.svg';
import { useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

export const ErrorPage = () => {
  const history = useHistory();
  const { t } = useTranslation();
  return (
    <Panel cx={styles.wrapper}>
      <img src={ErrorIcon} />
      <Text>{t('notFoundPage.title')}</Text>
      <Button
        color="accent"
        caption={t('notFoundPage.cta')}
        onClick={() => history.push('/')}
      />
    </Panel>
  );
};
