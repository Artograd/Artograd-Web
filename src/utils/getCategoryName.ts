import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPage.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPage.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPage.newTender.categories.murals' },
  { id: 3, name: 'tendersPage.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPage.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPage.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPage.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPage.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPage.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPage.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
