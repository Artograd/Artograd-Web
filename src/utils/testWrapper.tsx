import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { store } from '../store/store';

export const testWrapper = async ({
  component,
  history,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  
}) => {
  return renderWithContextAsync(<Provider store={store}><Router history={history}>{component}</Router></Provider>);
};
