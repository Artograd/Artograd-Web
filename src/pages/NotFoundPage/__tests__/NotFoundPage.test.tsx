import { ErrorPage } from '../NotFoundPage';
import { createMemoryHistory } from 'history';
import { screen, fireEvent } from '@epam/uui-test-utils';
import { testWrapper } from '../../../utils/testWrapper';

describe('Not found page', () => {
  const history = createMemoryHistory();
  test('renders correctly', async () => {
    const component = await testWrapper({ component: <ErrorPage />, history });

    expect(component).toMatchSnapshot();
  });

  test('redirect on click page CTA', async () => {
    await testWrapper({ component: <ErrorPage />, history });
    fireEvent.click(screen.getByText('Return to homepage'));

    expect(history.location.pathname).toBe('/');
  });
});
