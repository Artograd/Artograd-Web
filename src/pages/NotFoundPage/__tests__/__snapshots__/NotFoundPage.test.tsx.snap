// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel Dy3NM- wrapper _54i0RB"
      >
        <img
          src="error.svg"
        />
        <div
          class="ZTInfz uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography NTnjYI oep1Ot QL0z0q En3MXG"
        >
          Page not found!
        </div>
        <button
          class="uui-button-box uui-enabled -clickable EBheCm pSu6kE uui-button uui-fill-solid uui-color-accent uui-size-36"
          data-testid="not-found-cta"
          tabindex="0"
          type="button"
        >
          <div
            class="uui-caption"
          >
            Return to homepage
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel Dy3NM- wrapper _54i0RB"
    >
      <img
        src="error.svg"
      />
      <div
        class="ZTInfz uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography NTnjYI oep1Ot QL0z0q En3MXG"
      >
        Page not found!
      </div>
      <button
        class="uui-button-box uui-enabled -clickable EBheCm pSu6kE uui-button uui-fill-solid uui-color-accent uui-size-36"
        data-testid="not-found-cta"
        tabindex="0"
        type="button"
      >
        <div
          class="uui-caption"
        >
          Return to homepage
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
